applications/tari_console_wallet/src/automation/command_parser.rs

```diff
@@ -0,0 +1,208 @@
+        .next()
+        .ok_or_else(|| ParseError::Empty("start amount".to_string()))?;
+    let start_amount = MicroTari::from_str(start_amount)?;
+    parsed_args.push(ParsedArgument::Amount(start_amount));
+
+    // increment amount
+    let inc_amount = args
+        .next()
+        .ok_or_else(|| ParseError::Empty("increment amount".to_string()))?;
+    let inc_amount = MicroTari::from_str(inc_amount)?;
+    parsed_args.push(ParsedArgument::Amount(inc_amount));
+
+    // start time utc or 'now'
+    let start_time = args.next().ok_or_else(|| ParseError::Empty("start time".to_string()))?;
+    let now = Utc::now();
+    let start_time = if start_time != "now" {
+        parse_date_string(&start_time, now, Dialect::Uk).map_err(ParseError::Date)?
+    } else {
+        now
+    };
+    parsed_args.push(ParsedArgument::Date(start_time));
+
+    // public key/emoji id
+    let pubkey = args
+        .next()
+        .ok_or_else(|| ParseError::Empty("public key or emoji id".to_string()))?;
+    let pubkey = parse_emoji_id_or_public_key(pubkey).ok_or(ParseError::PublicKey)?;
+    parsed_args.push(ParsedArgument::PublicKey(pubkey));
+
+    // message
+    let message = args.collect::<Vec<&str>>().join(" ");
+    parsed_args.push(ParsedArgument::Text(message));
+
+    Ok(parsed_args)
+}
+
+fn parse_send_tari(mut args: SplitWhitespace) -> Result<Vec<ParsedArgument>, ParseError> {
+    let mut parsed_args = Vec::new();
+
+    // amount
+    let amount = args.next().ok_or_else(|| ParseError::Empty("amount".to_string()))?;
+    let amount = MicroTari::from_str(amount)?;
+    parsed_args.push(ParsedArgument::Amount(amount));
+
+    // public key/emoji id
+    let pubkey = args
+        .next()
+        .ok_or_else(|| ParseError::Empty("public key or emoji id".to_string()))?;
+    let pubkey = parse_emoji_id_or_public_key(pubkey).ok_or(ParseError::PublicKey)?;
+    parsed_args.push(ParsedArgument::PublicKey(pubkey));
+
+    // message
+    let message = args.collect::<Vec<&str>>().join(" ");
+    parsed_args.push(ParsedArgument::Text(message));
+
+    Ok(parsed_args)
+}
+
+#[test]
+fn test_parse_command() {
+    let command_str = "";
+    let parsed = parse_command(command_str);
+    assert!(parsed.is_err());
+
+    let command_str = "send-tari asdf";
+    let parsed = parse_command(command_str);
+    assert!(parsed.is_err());
+
+    let command_str = "send-tari 999T";
+    let parsed = parse_command(command_str);
+    assert!(parsed.is_err());
+
+    let command_str = "send-tari 999T asdf";
+    let parsed = parse_command(command_str);
+    assert!(parsed.is_err());
+
+    let command_str = "send-tari 999T 6a493210f7499cd17fecb510ae0cea23a110e8d5b901f8acadd3095c73a3b919";
+    let parsed = parse_command(command_str);
+    assert!(parsed.is_ok());
+
+    let command_str = "send-tari 999T 6a493210f7499cd17fecb510ae0cea23a110e8d5b901f8acadd3095c73a3b919 message";
+    let parsed = parse_command(command_str);
+    assert!(parsed.is_ok());
```

Comment:
Nit: actually test that the parsed values are correct and also test uT and no unit on the tari amount.

Reply:
Agreed, and I'd like to add the other commands too